Dockerfile

```diff
@@ -1,76 +1,56 @@
-# Create common base stage
-FROM python:3.6-slim as base
-
-WORKDIR /build
-
-# Create virtualenv to isolate builds
-RUN python -m venv /build
-
-# Install common libraries
-RUN apt-get update -qq \
- && apt-get install -y --no-install-recommends \
-    # required by psycopg2 at build and runtime
-    libpq-dev \
-     # required for health check
-    curl \
- && apt-get autoremove -y
-
-# Make sure we use the virtualenv
-ENV PATH="/build/bin:$PATH"
-
-# Stage to build and install everything
-FROM base as builder
-
-WORKDIR /src
-
-# Install all required build libraries
-RUN apt-get update -qq \
- && apt-get install -y --no-install-recommends \
-    build-essential \
-    wget \
-    openssh-client \
-    graphviz-dev \
-    pkg-config \
-    git-core \
-    openssl \
-    libssl-dev \
-    libffi6 \
-    libffi-dev \
-    libpng-dev
-
-# Copy only what we really need
-COPY README.md .
-COPY setup.py .
-COPY setup.cfg .
-COPY MANIFEST.in .
-COPY requirements.txt .
-COPY LICENSE.txt .
-
-# Install dependencies
-RUN pip install -U pip && pip install --no-cache-dir -r requirements.txt
-
-# Install Rasa as package
-COPY rasa ./rasa
-RUN pip install .[sql]
-
-# Runtime stage which uses the virtualenv which we built in the previous stage
-FROM base AS runner
-
-# Copy virtualenv from previous stage
-COPY --from=builder /build /build
-
-WORKDIR /app
+FROM python:3.6-slim as python_builder
+RUN apt-get update -qq && \
+  apt-get install -y --no-install-recommends \
+  build-essential \
+  wget \
+  openssh-client \
+  graphviz-dev \
+  pkg-config \
+  git-core \
+  openssl \
+  libssl-dev \
+  libffi6 \
+  libffi-dev \
+  libpng-dev \
+  # required by psycopg2 at build and runtime
+  libpq-dev \
+  curl
+
+# install poetry
+# keep this in sync with the version in pyproject.toml and Dockerfile
+ENV POETRY_VERSION 1.0.3
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+ENV PATH "/root/.poetry/bin:/opt/venv/bin:${PATH}"
+
+# copy files
+COPY . /app/
+
+# install dependencies
+RUN python -m venv /opt/venv && \
+  . /opt/venv/bin/activate && \
+  pip install --no-cache-dir -U 'pip<20' && \
+  cd /app && \
+  poetry install --no-dev --no-interaction
+
+# start a new build stage
+FROM python:3.6-slim
+
+# copy everything from /opt
+COPY --from=python_builder /opt/venv /opt/venv
+COPY --from=python_builder /app /app
```

Comment:
why do we need this one?

Reply:
what do you mean? to copy some specific directories from builder

Reply:
Yep, but what's in there which we need in this stage? (and why do we need it)

Reply:
You're right — removed it.